### PR TITLE
[minifb] Bump to 2023-02-03

### PR DESCRIPTION
**Pull request by xiaozhuai**
1. Bump to 2023-02-03
2. Fix install
3. Add license
4. Modernization

### CODE REVIEW

**Comment by xiaozhuai** [comment]
Tested usage

```cmake
cmake_minimum_required(VERSION 3.2)
project(test_minifb)

set(CMAKE_CXX_STANDARD 14)

find_package(minifb CONFIG REQUIRED)

add_executable(test_minifb main.cpp)

target_link_libraries(test_minifb PRIVATE minifb::minifb)
```

```cpp
#include <iostream>
#include <MiniFB_cpp.h>
#include <cmath>

int main() {
    const int windowWidth = 1280;
    const int windowHeight = 960;
    auto *window = mfb_open_ex("minifb", windowWidth, windowHeight, 0);

    float scaleX, scaleY;
    mfb_get_monitor_scale(window, &scaleX, &scaleY);

    const int width = (int) round(float(windowWidth) * scaleX);
    const int height = (int) round(float(windowHeight) * scaleY);

    auto *image = new uint32_t[width * height];

    do {
        for (int i = 0; i < width * height; ++i) {
            image[i] = rand();
        }
        int state = mfb_update_ex(window, image, width, height);

        if (state < 0) {
            window = nullptr;
            break;
        }
    } while (mfb_wait_sync(window));

    delete[] image;

    return 0;
}
```

**Comment by xiaozhuai** [comment]
Also open a pr on upstream https://github.com/emoon/minifb/pull/102.
Please wait before merging this pr.
If upstream willing to take the change, then we can remove `fix-install.patch` file.

File: ports/minifb/fix-install.patch [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+-    add_compile_options("$<IF:$<CONFIG:Debug>,/MDd,/MD>")
+     # Program database for edit and continue
+     add_compile_options("$<IF:$<CONFIG:Debug>,/ZI,/Zi>")
+     # Optimizations
+@@ -256,14 +254,14 @@ add_library(minifb STATIC
+ if(APPLE)
+ 
+     if(IOS)
+-        target_link_libraries(minifb
++        target_link_libraries(minifb PUBLIC
```
Comment (dg0yt) [inline ports/minifb/fix-install.patch:19]: Which of these and the following libs are really public (i.e. affecting users at compile-time, not link-time)?

File: ports/minifb/fix-install.patch [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+     set_property(TARGET hidpi PROPERTY FOLDER "Tests")
+ endif()
+ 
++add_library(minifb::minifb ALIAS minifb)
++include(GNUInstallDirs)
++target_include_directories(minifb INTERFACE $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
++install(TARGETS minifb EXPORT minifb)
++file(GLOB_RECURSE HEADERS "${CMAKE_CURRENT_LIST_DIR}/include/*.h")
++install(FILES ${HEADERS} DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}")
++install(EXPORT minifb FILE minifb-config.cmake DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/minifb" NAMESPACE minifb::)
```
Comment (dg0yt) [inline ports/minifb/fix-install.patch:90]: Unless exported config comes from upstream, it must be in the `unofficial::<Pkg>::` namespace.
If upstream is not willing to accept that change, this PR must be changed.

**Comment by BillyONeal** [comment]
I manually checked that the SHA in here is not affected by https://github.com/microsoft/vcpkg/issues/29288

File: ports/minifb/fix-install.patch (hunk 1 above)
Comment (xiaozhuai) [reply ports/minifb/fix-install.patch:19]: All of these affecting users at link-time and should be public.

File: ports/minifb/fix-install.patch (hunk 2 above)
Comment (xiaozhuai) [reply ports/minifb/fix-install.patch:90]: Sure~

File: ports/minifb/fix-install.patch (hunk 1 above)
Comment (dg0yt) [reply ports/minifb/fix-install.patch:19]: > All of these affecting users at link-time and should be public.

This doesn't answer my question, but repeats the reaon why I ask. Link-time means `PRIVATE`. This is the minimum and preferred. Only if the dependencies' include dirs or macros are needed for using minifb's API at compile-time, then we are at `PUBLIC`.

Comment (xiaozhuai) [reply ports/minifb/fix-install.patch]: > Link-time means `PRIVATE`. This is the minimum and preferred. Only if the dependencies' include dirs or macros are needed for using minifb's API at compile-time, then we are at `PUBLIC`.

You are right. It's my mistake, all these lib should be private.

**Comment by xiaozhuai** [comment]
@MonicaLiu0311 @dg0yt Since upstream has accepted https://github.com/emoon/minifb/pull/102, I've update this pr, it's ok to be merged now.

**Review by MonicaLiu0311 — DISMISSED** [review]

**Review by BillyONeal — APPROVED** [review]
Thanks for the update!

**Pull request merged**